plugin/src/test/java/io/jenkins/plugins/forensics/git/delta/GitDeltaCalculatorITest.java

```diff
@@ -0,0 +1,334 @@
+package io.jenkins.plugins.forensics.git.delta;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import edu.hm.hafner.util.FilteredLog;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import io.jenkins.plugins.forensics.delta.model.Change;
+import io.jenkins.plugins.forensics.delta.model.ChangeEditType;
+import io.jenkins.plugins.forensics.delta.model.Delta;
+import io.jenkins.plugins.forensics.delta.model.FileChanges;
+import io.jenkins.plugins.forensics.delta.model.FileEditType;
+import io.jenkins.plugins.forensics.git.delta.model.GitDelta;
+import io.jenkins.plugins.forensics.git.util.GitITest;
+
+import static io.jenkins.plugins.forensics.assertions.Assertions.*;
+
+/**
+ * Integration test for the class {@link GitDeltaCalculator}.
+ *
+ * @author Florian Orendi
+ */
+public class GitDeltaCalculatorITest extends GitITest {
+
+    /**
+     * The delta result should be empty if there are invalid commits.
+     */
+    @Test
+    public void shouldCreateEmptyDeltaIfCommitsAreInvalid() {
+        GitDeltaCalculator deltaCalculator = createDeltaCalculator();
+
+        FilteredLog log = createLog();
+        Optional<Delta> delta = deltaCalculator.calculateDelta("", "", log);
+
+        assertThat(delta).isEmpty();
+    }
+
+    /**
+     * The Git diff file should be created properly.
+     */
+    @Test
+    @SuppressFBWarnings(value = "BC_UNCONFIRMED_CAST_OF_RETURN_VALUE", justification = "The cast is confirmed via an assertion")
+    public void shouldCreateDiffFile() {
+        GitDeltaCalculator deltaCalculator = createDeltaCalculator();
+        FilteredLog log = createLog();
+
+        final String referenceCommit = getHead();
+        final String fileName = "newFile";
+        final String content = "content";
+        writeFile(fileName, content);
+        addFile(fileName);
+        commit("test");
+        final String currentCommit = getHead();
+
+        Optional<Delta> result = deltaCalculator.calculateDelta(currentCommit, referenceCommit, log);
+        assertThat(result).isNotEmpty();
+
+        Delta delta = result.get();
+        assertThat(delta).hasCurrentCommit(currentCommit);
+        assertThat(delta).hasReferenceCommit(referenceCommit);
+        assertThat(delta).isInstanceOf(GitDelta.class);
+
+        assertThat(((GitDelta) delta).getDiffFile()).isEqualTo("diff --git a/newFile b/newFile\n"
```

Comment:
It would be simpler to use `isInstanceOfSatisfying`:

https://github.com/jenkinsci/forensics-api-plugin/blob/master/src/test/java/io/jenkins/plugins/forensics/reference/ReferenceBuildTest.java#L58